packages/core/versioning/core-version-http-server/src/version_http_toolkit.ts

```diff
@@ -99,14 +100,25 @@ export interface VersionHTTPToolkit {
   ): VersionedRouter<Ctx>;
 }
 
+type WithRequiredProperty<Type, Key extends keyof Type> = Type & {
+  [Property in Key]-?: Type[Property];
+};
+
+/**
+ * Versioned route access flag, required
+ * - '/api/foo' is 'public'
+ * - '/internal/my-foo'  is 'internal'
+ * Required
+ */
+type VersionedRouteConfigOptions = WithRequiredProperty<RouteConfigOptions<RouteMethod>, 'access'>;
```

Comment:
One alternative:
```suggestion
type VersionedRouteConfigOptions = Required<Pick<RouteConfigOptions<RouteMethod>, 'access'>>;
```

Reply:
Using `Pick` doesn't allow us to specify `timeout`.

Reply:
👍🏻 . WDYT about moving your type helper to the type utilities package?